### DISCUSSION TITLE
Allow having a bucket name in the S3 URL

### DISCUSSION THREAD

**Discussion by regzon** [outline/outline#3606] [Self Hosting] [answered]
I manage a self-hosted Outline for my personal needs, and recently I've migrated from the version 0.52.0 up to the latest one. Commits 6fa760688bc3c0ab3a20883571adf90a405a150c and 86d9a14c5c6be7a5d967667a6ca8e9460c37686c made it possible to use Virtual Hosted-Style URLs for AWS S3 users, but made it impossible to use bucket names in URLs for users of self-hosted S3-compatible services.

For me, it broke attachments-management completely, and the time it took to find the issue is tremendous. In theory, it can prevent a lot of new users from using Outline (as I didn't find any warning or a note in the documentation, and the arising error doesn't give any hints on solving the issue).

From my point of view, this issue could be resolved by handling non-AWS URLs separately. The same as previously mentioned commits check for an occurrence of the bucket name in the URL, we can first check if the URL is actually AWS-related (by checking if the string `amazonaws.com` is included in the URL, or by matching the URL against a regular expression).

Related issue #2225 (stalled).

**Comment by tommoor** [comment, marked as answer]
I believe this refactor will help – https://github.com/outline/outline/pull/3277 – really should try and get it landed by next release.

**Reply by regzon** [reply]
Okay, thanks. I'll check it at the next release.
